integration-tests/utils.py

```diff
@@ -244,15 +244,20 @@ def all_units_ready(model):
         bundle_dir = os.path.join(temp_dir, 'bundle')
         cmd = ['charm', 'pull', '--channel', channel, url, bundle_dir]
         await asyncify(subprocess.check_call)(cmd)
+        data_path = os.path.join(bundle_dir, 'bundle.yaml')
+        with open(data_path) as f:
+            data = yaml.load(f)
         if snap_channel:
-            data_path = os.path.join(bundle_dir, 'bundle.yaml')
-            with open(data_path) as f:
-                data = yaml.load(f)
             for app in ['kubernetes-master', 'kubernetes-worker']:
                 options = data['services'][app].setdefault('options', {})
                 options['channel'] = snap_channel
-            with open(data_path, 'w') as f:
-                yaml.dump(data, f)
+        if await is_localhost():
+            if 'options' not in data['services']['kubernetes-worker']:
+                data['services'][app].setdefault('options', {})
+            options = data['services'][app]['options']
```

Comment:
Where does `app` come from? It looks like it comes from the `for` loop above, which is confusing, to say the least.

Reply:
You are right, it's kubernetes-worker